### PR TITLE
Np 46966 Update report queries to new views

### PR DESCRIPTION
**Pull request by monaullah**
Update report queries according to new views

Also, fixed some bugs on the way:
- In view of publication: Expect status as String (Not 100% sure here, added TODO to verify with brinxmat)
- In view of publication: Fix `departmentUri` and `facultyUri` bindings (Not 100% sure here, added TODO to verify with brinxmat)
- In view of nvi candidate: Allow optional values for non applicable candidates
- Updatet test data, adding NviCandidate `id` and moving funding `label` to funding source

### CODE REVIEW

File: data-report-commons/src/test/java/commons/ViewCompilerTest.java [hunk 1]
```diff
@@ -34,6 +46,15 @@ void shouldReduceTriplesToNviCandidateViewRequiredToProduceApiData() {
         Assertions.assertTrue(expected(NVI_CANDIDATE_NT, uri).isIsomorphicWith(model));
     }
 
+    @Test
+    void shouldReduceTriplesToNviCandidateViewRequiredToProduceApiDataWhenCandidateIsNotApplicable() {
+        var uri = randomUri();
+        var inputStream = StaticTestDataUtil.getNonApplicableNviCandidate(uri);
+        var model = new ViewCompiler(inputStream).extractView(uri);
+        Model expected = expected(NON_APPLICABLE_NVI_CANDIDATE_NT, uri);
```
Comment (brinxmat) [inline data-report-commons/src/test/java/commons/ViewCompilerTest.java:54]: var

File: report-api/src/main/resources/template/nvi.sparql [hunk 2]
```diff
@@ -21,73 +21,62 @@ WHERE {
...
+        :isApplicable false .
 
       FILTER(?modifiedDate >= "__AFTER__"^^xsd:dateTime)
       FILTER(?modifiedDate < "__BEFORE__"^^xsd:dateTime)
 
       BIND(STR(?publicationUri) AS ?publicationId)
-      BIND(STR(?isApplicableBoolean) AS ?isApplicable)
-
-      FILTER(?isApplicableBoolean = false)
+      BIND("false" AS ?isApplicable)
```
Comment (brinxmat) [inline report-api/src/main/resources/template/nvi.sparql:79]: Hmm. Probably a boolean, but we didn't create the datatype in the flattening?

File: report-api/src/main/resources/template/publication.sparql [hunk 3]
```diff
@@ -16,44 +16,38 @@ SELECT DISTINCT
...
-         :entityDescription ?entityDescription .
+         :year ?year ;
+         :mainTitle ?publicationTitle ;
+         :publicationContext ?publicationContext .
 
     FILTER(?modifiedDate >= "__AFTER__"^^xsd:dateTime)
     FILTER(?modifiedDate < "__BEFORE__"^^xsd:dateTime)
 
     BIND(STR(?uri) AS ?publicationId)
+    BIND(REPLACE(STR(?instanceType), "https://nva.sikt.no/ontology/publication#", "", "i") AS ?publicationCategory)
```
Comment (brinxmat) [inline report-api/src/main/resources/template/publication.sparql:31]: In a previous query, I notice we still use the complex regex variant…maybe we need to, but check usages of this as it is difficult to grok.

**Review by brinxmat — APPROVED** [review]

File: data-report-commons/src/main/resources/view_of_publication.sparql [hunk 4]
```diff
@@ -133,6 +135,10 @@ CONSTRUCT {
     ( :PUBLISHED "PUBLISHED" )
     ( :DELETED "DELETED" )
     ( :UNPUBLISHED "UNPUBLISHED" )
+    ##TODO: Clarify these with brinxmat
```
Comment (ketilaa) [inline data-report-commons/src/main/resources/view_of_publication.sparql:138]: Not sure what effect this will have? What is it you want to achieve?

**Review by ketilaa — APPROVED** [review]
Some comments. I did not understand everything, but did not see anything that did not make sense either. Approved.

**Pull request merged**